### PR TITLE
[GH-43] Use SiteUrl for calls from the webapp

### LINKED ISSUE #43: Use SiteURL for all calls from the webapp
If the server's SiteURL is configured with a subpath, the webapp's API calls do not work. The task here is to make this plugin's client to prefix its URLs with the SiteURL.

Here's an example of how the SiteURL may be computed:
https://github.com/mattermost/mattermost-plugin-jira/blob/19a9c2442817132b4eee5c77e259b80a40188a6a/webapp/src/selectors/index.js#L13-L26

### PR DESCRIPTION
**Pull request by sudiptog81**
#### Summary

Added `getServerRoute()` and updated the Client to get `SiteURL` from the client configuration in the state to support subpath deployments.

#### Ticket Link

Fixes #43.

### CODE REVIEW

File: webapp/src/client/client.js [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
+import {getSiteURL} from '../selectors';
 
 export default class Client {
-    constructor() {
-        this.url = `/plugins/${pluginId}/api/v1`;
-    }
-
-    getAttributes = async (userID = '') => {
-        return this.doGet(`${this.url}/attributes?user_id=` + userID);
+    getAttributes = async (state = {}, userID = '') => {
```
Comment (mickmister) [inline webapp/src/client/client.js:7]: Seems a bit weird to have this function explicitly take the whole state as one of its arguments. Can we instead set the `Client`'s server URL on plugin initialization, similar to how it's done in the zoom plugin? https://github.com/mattermost/mattermost-plugin-zoom/blob/0ba667b10ac437d63600686753b985476eb9bb95/webapp/src/index.js#L26

**Review by mickmister — CHANGES_REQUESTED** [review]
@sudiptog81 Thanks for your contribution :+1:  I have one request for changing where we set the siteURL, otherwise LGTM

File: webapp/src/client/client.js (hunk 1 above)
Comment (sudiptog81) [reply webapp/src/client/client.js:7]: Thank you for pointing to the right direction! Addressing this in the next commit.

**Review by mickmister — APPROVED** [review]
LGTM :+1: Thanks for the contribution @sudiptog81!

File: webapp/src/actions/index.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ export function getAttributes(userID = '') {
 
         let data = [];
         try {
-            data = await Client.getAttributes(userID);
+            data = await Client.getAttributes(getState(), userID);
```
Comment (sudiptog81) [inline webapp/src/actions/index.js:18]:
```suggestion
            data = await Client.getAttributes(userID);
```

**Review by marianunez — APPROVED** [review]
Looks great! Thanks @sudiptog81!

**Comment by DHaussermann** [comment]
Hi @sudiptog81 thank you for adding the subpath support on this 🎉 

I ran into snag testing this on a subpath server. I'm not able to configure the groups via the admin side UI. Searching for users or teams will return a 404 that seems like it's missing the subdomain. This seems like the very issue you've resolved. Is it possible we need the solution to also be applied to the admin UI or could something else be happening?

Screen Shot 2020-11-06 at 1 49 02 PM

**Comment by jasonblais** [comment]
@sudiptog81 let us know if you have any comments or feedback to @DHaussermann's question above :)

**Comment by larkox** [comment]
Hi @sudiptog81, do you think you could keep working on this issue?

**Comment by sudiptog81** [comment]
I'm pre-occupied with some coursework, closing this, sorry for not making it :)

**Comment by jasonblais** [comment]
Thanks for letting us know, hope to see you around the community in the future :) Big thanks for your help here and getting the work started!

**Pull request closed without merging**